File: Examples.VB/Features/Formatting/CellType/AddComboBoxCellType.vb
Namespace Features.Formatting.CellType
    Public Class AddComboBoxCellType
        Inherits ExampleBase

        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)
            Dim cellType As New ComboBoxCellType With {
                .EditorValueType = EditorValueType.Value
            }

            cellType.Items.Add(New ComboBoxCellItem With {
                .Value = "US",
                .Text = "United States"
            })

            cellType.Items.Add(New ComboBoxCellItem With {
                .Value = "CN",
                .Text = "China"
            })

            cellType.Items.Add(New ComboBoxCellItem With {
                .Value = "JP",
                .Text = "Japan"
            })

            With worksheet.Range!C5
                .CellType = cellType
                .Value = "CN"
            End With
        End Sub
    End Class
End Namespace

File: Examples/Features/Formatting/CellType/AddComboBoxCellType.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.Formatting.CellType
{
    public class AddComboBoxCellType : ExampleBase
    {
        public override bool IsNew => true;
        public override void Execute(Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];
            ComboBoxCellType cellType = new ComboBoxCellType
            {
                EditorValueType = EditorValueType.Value
            };
            ComboBoxCellItem item = new ComboBoxCellItem
            {
                Value = "US",
                Text = "United States"
            };
            cellType.Items.Add(item);
            item = new ComboBoxCellItem
            {
                Value = "CN",
                Text = "China"
            };
            cellType.Items.Add(item);
            item = new ComboBoxCellItem
            {
                Value = "JP",
                Text = "Japan"
            };
            cellType.Items.Add(item);

            worksheet.Range["C5"].CellType = cellType;
            worksheet.Range["C5"].Value = "CN";
        }
    }
}
